src/H5C.c

```diff
@@ -7854,39 +8386,42 @@ H5C_validate_pinned_entry_list(H5C_t *cache_ptr)
         (cache_ptr->pel_head_ptr != cache_ptr->pel_tail_ptr))
         HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "Check 1 failed")
 
+    if (cache_ptr->pel_len < 0)
+        HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "Check 2 failed")
```

Comment:
Same - Numbering these checks is pointless - the line # is included in the error output and the error check # has no semantic content on the error stack.   It also makes silly-work to update the #'s for existing checks when new errors are inserted.